contracts/examples/multisig/src/multisig_perform.rs

```diff
@@ -87,9 +87,24 @@ pub trait MultisigPerformModule: crate::multisig_state::MultisigStateModule {
         self.perform_action(action_id)
     }
 
+    #[event("actionPerformed")]
+    fn perform_action_event(
+        &self,
+        data: ActionFullInfo<Self::Api>,
+    );
```

Comment:
Serializing everything into a single object makes it hard to decode.
There should be separate events for each action type, especially for async calls. 
The action id is not relevant, you already have it in the call. You should focus on _what_ the action does.